### ISSUE TITLE
Variable existence checks stop working with array access

### ISSUE THREAD

**Issue by klx01** [kalessil/phpinspectionsea#1612]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.5 (and previous versions as well) |
| Language level | PHP 7.4 (and previous versions as well)                        |

Current behaviour
--
```php
// if a variable is not defined, I currently get this error, which is correct 
function foo1() {
    $a1 = $b1 ?? 1;     // [EA] '$b1' seems to be not defined in the scope.
    $a2 = isset($b2);   // [EA] '$b2' seems to be not defined in the scope.
    $a3 = empty($b3);   // [EA] '$b3' seems to be not defined in the scope.
}

// but if i add array offset access to that variable, I no longer get any errors
function foo2() {
    $a1 = $b1[0] ?? 1;     // no error
    $a2 = isset($b2[0]);   // no error
    $a3 = empty($b3[0]);   // no error
}
```

Expected behaviour
--
Expected behaviour is to get "Variable seems to be not defined in the scope" error in both cases

**Comment by ea-inspections-team** [comment]
Thanks for reporting @klx01 ! The mentioned cases detection is available in [Php Inspections (EA Ultimate) ](https://plugins.jetbrains.com/plugin/16935-php-inspections-ea-ultimate-)for a while already. Closing the issue.
